internal/services/machinelearning/machine_learning_datastore_datalake_gen1_resource.go

```diff
@@ -0,0 +1,355 @@
+package machinelearning
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/machinelearningservices/2022-05-01/datastore"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/machinelearningservices/2022-05-01/workspaces"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/tf"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/machinelearning/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type MachineLearningDataStoreDataLakeGen1 struct{}
```

Comment:
@xuzhang3 Gen1 is deprecated and we've removed support for this in 3.0 since you can no longer provision new accounts: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/guides/3.0-upgrade-guide#resource-azurerm_data_share_dataset_data_lake_gen1

https://learn.microsoft.com/en-gb/azure/storage/blobs/data-lake-storage-migrate-gen1-to-gen2

What's the use-case for adding this resource, given new instances can't be created and the Service is entirely removed next year?

Reply:
@tombuildsstuff this resource was generated based on the data store types. @deeikele  do we need to support gen1 resources as this service will be deprecated soon?

Reply:
@xuzhang3 No, there's no need to support gen1